### PR TITLE
HPCC-16731 Set MALLOC_ARENA_MAX for better glibc malloc performance

### PR DESCRIPTION
**Pull request by mckellyln**
@jakesmith please review.

NOTE: We could look into adding a setting to the environment.conf for this, if you wanted to make this more configurable.

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

### CODE REVIEW

File: initfiles/sbin/hpcc_setenv.in [hunk 1]
```diff
@@ -102,7 +102,16 @@ IFS="${OIFS}"
...
 
 # use less heap when threaded
-export MALLOC_ARENA_MAX=8
+# but if set too low it can affect performance significantly
+num_arenas=$(getconf _NPROCESSORS_ONLN 2>/dev/null)
+if [ -z "$num_arenas" ] ; then
+    num_arenas = 8
+elif [ $num_arenas -lt 8 ] ; then
+    num_arenas = 8
+elif [ $num_arenas  -gt 32 ] ; then
```
Comment (jakesmith) [inline initfiles/sbin/hpcc_setenv.in]: trivial: 2 spaces after num_arenas

**Comment by jakesmith** [comment]
Not worth making configurable I think, we expect TBB to be used and therefore these setting not to be used, but these new defaults will help anyone who's not using TBB.

@richardkchapman - ready to merge.

File: initfiles/sbin/hpcc_setenv.in (hunk 1 above)
Comment (mckellyln) [inline initfiles/sbin/hpcc_setenv.in]: Fixed

**Comment by richardkchapman** [comment]
The Jira says this should target 6.0.12 (and I think I agree)

**Comment by mckellyln** [comment]
Closing to rebase to 6.0.12

**Pull request closed without merging**